### PR TITLE
chore: cleanup project for development (AR-754)

### PR DESCRIPTION
**Pull request by vitorhugods**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-754" title="AR-754" target="_blank">TaskAR-754</a>  Cleanup AR for development of the new UI
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [X] contains a reference JIRA issue number like `SQPIT-764`
  - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

AR-754

### Solutions

* Renamed the current `app` module to `legacy` so we can more easily check the current reloaded app.
Cleaned up the `app` module so it's closer to a white canvas.
* Added `.editorconfig` to try to reduce conflicts due to IDE style changes.
* Added Github Actions for running tests and checking detekt lint

Currently, detekt and tests are running on Jenkins that's not accessible to everyone, so in order to make it transparent to all the collaborators, making these checks in GitHub sounds like a logical step forward.

### Testing

N/A

----
#### PR Post Merge Checklist for internal contributors

 - [X] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: settings.gradle.kts [hunk 1]
```diff
@@ -14,5 +14,6 @@ includeBuild("kalium") {
     // This dependency substitution should not be done on release mode once the Kalium library has been published to Maven repo
     dependencySubstitution {
         substitute(module("com.wire.kalium:kalium-network")).using(project(":network"))
```
Comment (gongracr) [inline settings.gradle.kts]: Shouldn't we need to remove the network module and leave only the `kalium-logic` as the only entry point from external consumers of Kalium?

Comment (vitorhugods) [reply settings.gradle.kts]: I guess we can. I just didn't want to check if `kalium-network` was being used when I passed through this file 😛 

I'll remove it

**Review by gongracr — APPROVED** [review]
Looks good enough to kick off to me 🧨

**Comment by vitorhugods** [comment]
Jcenter is now throwing some 502s. So I'll use admin rights to forcefully merge this and let everyone jump into the project.

**Pull request merged**